src/runner/test-run-controller.js

```diff
@@ -6,6 +6,18 @@ import TestRun from '../test-run';
 // Const
 const QUARANTINE_THRESHOLD = 3;
 
+class Quarantine {
+    constructor () {
+        this.passedTimes   = [];
```

Comment:
Now we store not only passed/failed test count, but also their indexes. So. let's name these properties `passedAttempts/failedAttempts`